omod/src/main/java/org/openmrs/module/referencedemodata/web/controller/DemoDataGenerationController.java

```diff
@@ -0,0 +1,96 @@
+package org.openmrs.module.referencedemodata.web.controller;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.referencedemodata.ReferenceDemoDataActivator;
+import org.openmrs.module.referencedemodata.ReferenceDemoDataConstants;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.openmrs.scheduler.TaskDefinition;
+import org.openmrs.scheduler.tasks.AbstractTask;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/" + ReferenceDemoDataConstants.REFERENCE_DEMO_DATA_URI)
+public class DemoDataGenerationController extends BaseRestController {
+	
+	private static ReferenceDemoDataActivator referenceDemoDataActivator;
+	
+	private static final String REFERENCE_DEMO_DATA_TASK_NAME = "Reference demo data generation task";
```

Comment:
There's an argument to be made here for having a ReferenceDemoDataRestConstants class or something to store these...

Reply:
These constants are more specific to this controller use-case. I'd have `ReferenceDemoDataRestConstants` in place when their re-usability is needed IMO. @ibacher do you envision some other features going to utilise these for now (in which case we rather address this now)?